### ISSUE TITLE
Add custom text-generation-inference support.

### ISSUE THREAD

**Issue by Yiannis128** [Yiannis128/esbmc-ai#59]
There should be an entry in `config.json` that allows for adding custom LLMs that are compatible with the `text-generation-inference` from HuggingFace.

```json
"example-llm": {
  "max_tokens": 1024,
  "url": "www.example.com",
  "config_message": "{history}\n{user_prompt}\n",
  "api_key_env": "EXAMPLE_KEY"
}
```

**Comment by Yiannis128** [comment]
This has been implemented in 446d663. Also created a [wiki](https://github.com/Yiannis128/esbmc-ai/wiki/AI-Models) page that describes how they work. Currently, it uses the HUGGINGFACE key for interactions.

**Comment by Yiannis128** [comment]
The `api_key_env` feature will be completed in https://github.com/Yiannis128/esbmc-ai/issues/60
